### PR TITLE
fix(#148): entity identity is embedding-based, never name-based

### PR DESCRIPTION
**Pull request by c-daly**
Closes #148.

## What
Entity de-duplication is decided in embedding space, not by the literal name string.

- **`hcg_client.add_node`**: when `uuid` is `None`, always mint a fresh `uuid4` — removed the `name+type` MATCH/reuse. Folding two distinct entities that happen to share a name is a lossy, text-based merge; duplicates are recoverable, false merges are not. (15/16 callers pass an explicit uuid; only the ingest path is affected.)
- **`proposal_processor`**: in-process (2a-pre) embedding dedup against siblings created earlier in the same ingest — the Milvus index is only flushed after the node loop, so the persisted search can't catch same-batch duplicates. Squared-L2 < `ENTITY_MATCH_THRESHOLD`.

Homonyms collapsing is an accepted edge case of the embedding resolver (Sophia is non-linguistic; an embedding is a vector, not text).

## Tests
- Rewrote `test_node_dedup` → `TestNodeIdentity`: same name+type → **distinct** uuids, no name lookup; explicit uuid → single MERGE.
- Updated `test_client_wrapper` (no dedup lookup).
- New live-Neo4j integration test: two same-name entities → two nodes; explicit uuid → one node.
- Full `tests/unit` green (315 passed); ruff/black/mypy clean.

### CODE REVIEW

File: src/sophia/ingestion/proposal_processor.py [hunk 1]
```diff
@@ -281,6 +291,41 @@ def _search_collection(
...
+                        best_uuid: str | None = None
+                        best_dist = ENTITY_MATCH_THRESHOLD
+                        for _coll, pending in pending_embeddings.items():
+                            for pending_node in pending:
+                                dist = _squared_l2_distance(
+                                    embedding, pending_node["embedding"]
+                                )
+                                if dist < best_dist:
+                                    best_dist = dist
+                                    best_uuid = pending_node["uuid"]
```
[the remark this answers is not in the document]
Comment (c-daly) [reply src/sophia/ingestion/proposal_processor.py]: Fixed in 0d07640 — the in-process dedup now iterates `pending_embeddings.get(collection, [])` only, matching the collection-scoped Milvus 2a search.

File: src/sophia/ingestion/proposal_processor.py [hunk 2]
```diff
@@ -281,6 +291,41 @@ def _search_collection(
...
+                            logger.info(
+                                "Entity '%s' matches a sibling created earlier in "
+                                "this ingest (%s, L2=%.3f); reusing, skipping "
+                                "creation",
+                                name,
+                                best_uuid,
+                                best_dist,
+                            )
+                            name_to_uuid[name] = best_uuid
+                            continue
```
[the remark this answers is not in the document]
Comment (c-daly) [reply src/sophia/ingestion/proposal_processor.py:331]: Fixed in 0d07640 — scoped the in-process dedup to the current collection. Added a regression test (similar embeddings in Entity vs Concept stay distinct) that fails on the old all-collections loop and passes after scoping, so batch membership no longer changes whether two nodes collapse.

File: src/sophia/ingestion/proposal_processor.py [hunk 3]
```diff
@@ -110,6 +110,16 @@ def _collection_for(node_type: str) -> str:
...
 
 
+def _squared_l2_distance(a: list[float], b: list[float]) -> float:
+    """Squared Euclidean (L2) distance between two vectors.
+
+    Matches Milvus' L2 metric (which returns squared distance), so distances
+    computed here are directly comparable to ``ENTITY_MATCH_THRESHOLD`` and to
+    ``search_similar`` scores. Lower is more similar.
+    """
+    return sum((x - y) ** 2 for x, y in zip(a, b))
```
[the remark this answers is not in the document]
Comment (c-daly) [reply src/sophia/ingestion/proposal_processor.py]: Done in 0d07640 — `zip(a, b, strict=True)`. A dimension mismatch now raises instead of silently truncating. (The collection-scoping fix also keeps compared vectors within one collection/model, so equal dims are expected.)
